systemd/google-cloud-ops-agent.service

```diff
@@ -17,11 +17,10 @@ Description=Google Cloud Ops Agent
 Wants=google-cloud-ops-agent-fluent-bit.service google-cloud-ops-agent-opentelemetry-collector.service
 
 [Service]
-Type=oneshot
+Type=simple
```

Comment:
Why are we doing this instead of relying on the OT Collector to write the metrics?

Reply:
The decision was taken due to the following arguments :
- If the OTel collector goes down, we can keep sending this metrics (also if there other health related metrics).
- A push to have the main service have more responsibility.

Still, it is a big change in the architecture, so there are a couple drawbacks.

I agree with the `simple` argument, `systemd` needs someway to know about the validation to correctly manage the subagents.

Reply:
Re:`Why are we doing this instead of relying on the OT Collector to write the metrics?`
### There are two parts for implementing this metrics.
### Part 1 - parsing and composing metrics
Parsing the config.yaml and composing the enabled_receivers timeseries shall really be done by the ops-agent binary rather the otel-ops-collector i think? `otel-ops-collector` as of today, it's a separate upstream repo that's not bundled with ops-agent(thought ops-agent seems the only one using it).  Thus there must be a binary running whenever ops-agent gets restarted that parses and composes this metric. Also this enabled_receiver metric is a gauge, thus it needs to be sent out every x interval. Therefore, it needs to maintain that running and ships this timeseries every x interval.

### Part two - send metrics
This PR is sending metrics directly into Monitoring API. One of the reason is this diagnostic metrics is tied to the config itself rather than any subagents, so whether the sub agents run or not shall not affect this metric being sent to the monitoring API.

Also grab @braydonk here who also mentioned using OT collector on writing the metrics. I always wonder what's the implied implementation for this option. How to achieve it.

Reply:
I'm suggesting that you calculate what the timeseries should be as part of config generation and bake it into the generated collector config, relying on the collector to do what it's designed to do and report the metrics upstream. e.g. it looks like you could use `otlpjsonfilereceiver` to accomplish this.

This PR reimplements a bunch of what the collector does, which means you're potentially introducing new and different bugs, and now you need to keep it in sync with whatever the collector is doing.

Reply:
re: `what's the implied implementation for this option`

There's two options I see:

**Option 1: Ops Agent Receiver**

We'd make a configurable `opsagentreceiver` in `opentelemetry-operations-collector` that reads the Ops Agent config on the system and does similar calculations as here.

**Option 2: Static Value Receiver**

We could make a `staticvaluereceiver` that is configured to emit a set of values every collection interval. This would allow us to calculate the values and embed them in the otel config.

Personally I'm partial to option 1, since it's more in line with the way otel receivers are usually designed, but I'm open to dicussion on that (GitHub comments is probably not the best medium for those discussions though, we can summarize on this thread when we discuss more holistically elsewhere).

Reply:
What's wrong with the existiing `otlpjsonfilereceiver` in lieu of option 2?

We should not do option 1 as-is because it's important that the configuration only gets read and parsed once. Otherwise what the metrics report as the running config might not match what's actually running (e.g. if the user edits the config files). So whatever we report from the collector, it needs to come from information that was assembled by the config generation. Whether we use `otlpjsonfilereceiver` or build our own generic receiver or build our own specific receiver I don't think matters much, but minimizing the code complexity is a valuable goal.

Reply:
From all the options using the opentelemetry collector, the one using `otlpjsonfilereceiver` is probably the simplest, straightforward and with minimum code complexity. The only hypothetical downside is that this sets values statically, but for our current "new metrics" (enabled_receivers, feature_tracking) this is not an issue. 

I've already synced with @braydonk about it and now I have a clear idea on how to implement it. I documented all this opinions and options in the design document. Still looking if it's possible to use the Opentelemetry SDK.